src/util/streamable_generator.h

```diff
@@ -126,7 +140,7 @@ class [[nodiscard]] basic_stream_generator {
   basic_stream_generator() : basic_stream_generator(noOpGenerator()){};
 
   basic_stream_generator(basic_stream_generator&& other) noexcept
-      : _coroutine{other._coroutine} {
+      : _method{other._method}, _coroutine{other._coroutine} {
```

Comment:
I was just confused:
Is the following correct:
We do not need to use `setContentEncoding` here, because the content encoding `_method` already is already correctly set up in the promise_type of the moved-in `_coroutine`?
I would add a short comment for this, as I was confused about this.

Reply:
Yes, that should be correct. But I don't quit understand where your confusion is coming from. This just behaves as a regular move constructor: `other` has to be a valid object and `this` just steals all the values from it, leaving them as-is, just nulling out the "pointer"